api/src/main/java/org/pmiops/workbench/api/DiskController.java

```diff
@@ -0,0 +1,117 @@
+package org.pmiops.workbench.api;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.logging.Logger;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.pmiops.workbench.leonardo.model.LeonardoGetPersistentDiskResponse;
+import org.pmiops.workbench.leonardo.model.LeonardoListPersistentDiskResponse;
+import org.pmiops.workbench.leonardo.model.LeonardoRuntimeStatus;
+import org.pmiops.workbench.model.Disk;
+import org.pmiops.workbench.model.EmptyResponse;
+import org.pmiops.workbench.model.WorkspaceAccessLevel;
+import org.pmiops.workbench.notebooks.LeonardoNotebooksClient;
+import org.pmiops.workbench.utils.mappers.LeonardoMapper;
+import org.pmiops.workbench.workspaces.WorkspaceAuthService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class DiskController implements DiskApiDelegate {
+  private static final Logger log = Logger.getLogger(DiskController.class.getName());
+  private final LeonardoNotebooksClient leonardoNotebooksClient;
+  private final WorkspaceDao workspaceDao;
+  private final WorkspaceAuthService workspaceAuthService;
+  private final LeonardoMapper leonardoMapper;
+
+  @Autowired
+  public DiskController(
+      LeonardoNotebooksClient leonardoNotebooksClient,
+      WorkspaceAuthService workspaceAuthService,
+      WorkspaceDao workspaceDao,
+      LeonardoMapper leonardoMapper) {
+    this.leonardoNotebooksClient = leonardoNotebooksClient;
+    this.workspaceAuthService = workspaceAuthService;
+    this.workspaceDao = workspaceDao;
+    this.leonardoMapper = leonardoMapper;
+  }
+
+  private DbWorkspace lookupWorkspace(String workspaceNamespace)
+      throws org.pmiops.workbench.exceptions.NotFoundException {
+    return workspaceDao
+        .getByNamespace(workspaceNamespace)
+        .orElseThrow(() -> new NotFoundException("Workspace not found: " + workspaceNamespace));
+  }
+
+  @Override
+  public ResponseEntity<Disk> getDisk(String workspaceNamespace) {
+    DbWorkspace dbWorkspace = lookupWorkspace(workspaceNamespace);
+    String googleProject = dbWorkspace.getGoogleProject();
+    try {
+      LeonardoGetPersistentDiskResponse response;
+      List<LeonardoListPersistentDiskResponse> responseList =
+          leonardoNotebooksClient.listPersistentDiskByProject(googleProject, false);
+      responseList.sort(
```

Comment:
why we want to sort the response?

Reply:
Because we want to get the latest created disk, in case of multiple existing disks.

Reply:
How you guarantee the last created disk is the one we want? 10 users are using the same workspace, and create PDs at the same time.  What would happen then?

Reply:
Self-answered from your doc: 
`Each user could only have one PD for one workspace and a PD is globally unique under the namespace <workspace, userID>.
`
So if that is unique per user per workspace. Do you need the logic to iterate the the response list and find the name matches all-of-us-**user-id**-uuid?
`user id` is available if you inject userProvider

Reply:
Ah, the doc is not updated. Some points here:

- By introducing UUID, the PD is not unique under <workspace, userID>. It's unique under <workspace, userID, UUID>.
- All the PDs under one workspace/google project are exclusively belong to one user.

So we could just pick the latest one from the result set returned by listPersistentDiskByProject(). We don't need to check if it's belong to the user anymore.

Reply:
> How you guarantee the last created disk is the one we want? 10 users are using the same workspace, and create PDs at the same time. What would happen then?

1. The workspace is exclusively used by one user. So it won't happen that multiple users creating multiple PDs at the same time
2. But multiple PDs would be created by the same user when
    - Directly calling the backend API to create PD
    - Trigger creating on multiple devices/browsers at the same time before the cache refreshed

Unfortuantely, we can't perfectly solve the second case for now. Because when we want to keep the uniqueness of PD while creating a new one with a unique UUID in a "copy-on-write" way. We need an atomic/transactional operation to guarantee the "compare-and-set/checking-then-creating" operation. But Leo doesn't provide it.

But we could still use some alternative way to decrease the risk of creating multiple PDs somehow.
- Call getPD API to check the PD existence before Creating a new one. As @calbach mentioned in the UUID PR.
- Use a cron job to monitor multiple PDs existence. (previously discussed with CH )

Reply:
> Ah, the doc is not updated. Some points here:
> 
> * By introducing UUID, the PD is not unique under <workspace, userID>. It's unique under <workspace, userID, UUID>.
> * All the PDs under one workspace/google project are exclusively belong to one user.
> 
> So we could just pick the latest one from the result set returned by listPersistentDiskByProject(). We don't need to check if it's belong to the user anymore.

This problem truly exits. Now it's fixed by filtering result sets with userID prefix in DiskController.getDisk().

Reply:
> Self-answered from your doc:
> `Each user could only have one PD for one workspace and a PD is globally unique under the namespace <workspace, userID>. `
> So if that is unique per user per workspace. Do you need the logic to iterate the the response list and find the name matches all-of-us-**user-id**-uuid?
> `user id` is available if you inject userProvider

This problem truly exits! Now it's fixed by filtering result sets with userID prefix in DiskController.getDisk().